### PR TITLE
Remove redundant WrappedConnection

### PR DESCRIPTION
**Pull request by richvdh**
The matrix federation client uses an HTTP connection pool, which times out its idle HTTP connections, so there is no need for any of this business.

### CODE REVIEW

File: synapse/http/matrixfederationclient.py [hunk 1]
```diff
@@ -320,23 +320,23 @@ def _send_request(
                         url_str,
                     )
 
-                    # we don't want all the fancy cookie and redirect handling that
-                    # treq.request gives: just use the raw Agent.
-                    request_deferred = self.agent.request(
```
Comment (richvdh) [inline synapse/http/matrixfederationclient.py]: for some reason, the change makes this now throw some exceptions synchronously, hence moving it inside the try/catch. It should have been there anyway imho.

File: tests/http/test_fedclient.py [hunk 2]
```diff
@@ -63,7 +66,7 @@ def test_client_never_connect(self):
         self.pump()
 
         # Nothing happened yet
-        self.assertFalse(d.called)
+        self.assertNoResult(d)
```
Comment (richvdh) [inline tests/http/test_fedclient.py]: this gives a slightly more helpful error when it fails.

File: tests/http/test_fedclient.py [hunk 3]
```diff
@@ -195,3 +198,63 @@ def test_client_sends_body(self):
...
+        self.assertFalse(conn.lostConnection)
+
+        # wait for a while
+        self.pump(120)
+
+        self.assertTrue(conn.lostConnection)
+
+
+@implementer(ITransport)
+class MockTransport(object):
```
Comment (hawkowl) [inline tests/http/test_fedclient.py]: Maybe https://twistedmatrix.com/documents/current/api/twisted.test.proto_helpers.StringTransport.html would reduce some duplication here? (`twisted.test.proto_helpers` is public API)

Comment (richvdh) [reply tests/http/test_fedclient.py]: yes :)

**Comment by richvdh** [comment]
@hawkowl: I was wondering about adding TLS into this, so that we could test the behaviour including the TLS layer, so started looking at Factories and Transports and Protocols, and soon found myself in a maze of twisty little passages, all alike. Have you any insights as to how one might do that?

**Pull request merged**